src/primitive.cc

```diff
@@ -61,6 +61,7 @@ Object* Primitive::allocate_large_integer(int64 value, Process* process) {
 
 Object* Primitive::allocate_array(int length, Object* filler, Process* process) {
   ASSERT(length <= Array::max_length_in_process());
```

Comment:
Drop this assert now?

Reply:
Returning null is safe and will cause an OOM instead of a vulnerability, but I still would prefer the assert if this happens while I am developing - the check should be in the calling primitive where we can return an appropriate exception.